xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-security/xwiki-platform-extension-security-index/src/main/java/org/xwiki/extension/security/internal/DefaultReviewsFetcher.java

```diff
@@ -62,25 +61,19 @@ public class DefaultReviewsFetcher implements ReviewsFetcher
     @Inject
     private Logger logger;
 
+    @Inject
+    private HttpClientFactory httpClientFactory;
+
     @Override
     public Optional<ReviewsMap> fetch() throws ExtensionSecurityException
     {
-        HttpRequest request = HttpRequest.newBuilder()
-            .uri(buildURI())
-            .GET()
-            .build();
-
-        try {
-            HttpResponse<String> response = HttpClient.newHttpClient().send(request, ofString());
-            ReviewsMap reviewsMap = new ObjectMapper().readValue(response.body(), ReviewsMap.class);
+        try (CloseableHttpClient httpClient = this.httpClientFactory.createHttpClientBuilder(Map.of()).build()) {
+            HttpGet getMethod = new HttpGet(buildURI());
+            CloseableHttpResponse execute = httpClient.execute(getMethod);
+            ReviewsMap reviewsMap = new ObjectMapper().readValue(execute.getEntity().getContent(), ReviewsMap.class);
```

Comment:
You might want to check if the request succeeded before reading it.

Reply:
done